src/lib.rs

```diff
@@ -254,98 +227,104 @@ impl<'a> Drop for Scope<'a> {
         }
 
         let recv = self.recv.take().unwrap();
-        let n = recv.wait().next();
+
+        let n = match Handle::try_current() {
```

Comment:
Why not just use `self.handle` for this?

Reply:
Good point. I'll do that.